### PR TITLE
feat: add subdomain/hostname to pods created

### PR DESCRIPTION
**Pull request by PFC-developer**
This PR adds hostname & subdomain to pods on creation (similar to how stateful sets do).
This is needed for a headless service to have podname.servicename... DNS resolution (instead of just servicename).
ala
`
root@harpoon-horcrux-1:/# host harpoon-0.harpoon
harpoon-0.harpoon.cosmos.svc.cluster.local has address 10.0.9.240

root@harpoon-horcrux-1:/# host harpoon-1.harpoon
harpoon-1.harpoon.cosmos.svc.cluster.local has address 10.0.7.151

root@harpoon-horcrux-1:/# host harpoon-2.harpoon
harpoon-2.harpoon.cosmos.svc.cluster.local has address 10.0.8.192

root@harpoon-horcrux-1:/# host harpoon
harpoon.cosmos.svc.cluster.local has address 10.0.7.151
harpoon.cosmos.svc.cluster.local has address 10.0.9.240
harpoon.cosmos.svc.cluster.local has address 10.0.8.192
`
before this patch only the last 'DNS' lookup would work.
I don't believe it has any negative effects

### CODE REVIEW

File: internal/fullnode/build_pods.go [hunk 1]
```diff
@@ -22,6 +22,8 @@ func BuildPods(crd *cosmosv1.CosmosFullNode, cksums ConfigChecksums) ([]diff.Res
 		if err != nil {
 			return nil, err
 		}
+		pod.Spec.Hostname = pod.Name
+		pod.Spec.Subdomain = crd.Name
```
Comment (DavidNix) [inline internal/fullnode/build_pods.go:26]: Please move to pod_builder.go. Otherwise, add unit test coverage showing why these changes are needed.

**Review by DavidNix — CHANGES_REQUESTED** [review]
This is great, thank you. 

All changes require unit test coverage before merge. See CONTRIBUTING.md. Please look at `pod_builder_test.go` and add tests there.

**Comment by PFC-developer** [comment]
moved logic to pod_builder and added test.

**Review by PFC-developer — COMMENTED** [review]
changes applied

**Review by DavidNix — COMMENTED** [review]
Thanks!

**Review by DavidNix — APPROVED** [review]
